File: ios/Classes/Cache.swift
import Foundation

final public class Cache {
    private init() {}
    
    public static let shared = Cache()
    
    private (set) public var latestEvent: [String: Any?]?
    
    public func updateLatestEvent(action: String, data: [String: Any?]) {
        latestEvent = [
            "event": action,
            "body": data,
        ]
    }
    
    public func clearLatestEvent() {
        latestEvent = nil
    }
}

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/Cache.kt
package com.hiennv.flutter_callkit_incoming

object Cache {
    var latestEvent: Map<String, Any>? = null
        private set

    fun updateLatestEvent(action: String, data: Map<String, Any>) {
        latestEvent = mapOf(
            "event" to action,
            "body" to data,
        )
    }

    fun clearLatestEvent() {
        latestEvent = null
    }
}
